File: skills/10-stages-developing/templates/javascript/function-template.js
/**
 * [Brief description of what this module does]
 * @module module-name
 */

/**
 * [Brief description of what the function does].
 *
 * @param {string} param1 - [Description of param1]
 * @param {number} param2 - [Description of param2]
 * @returns {{key1: string, key2: number}} Object containing processed results
 * @throws {Error} If param1 is empty or param2 is negative
 *
 * @example
 * const result = functionName('example', 42);
 * // Returns: { key1: 'processed_example', key2: 84 }
 */
function functionName(param1, param2) {
    // Input validation
    if (!param1) {
        throw new Error('param1 cannot be empty');
    }

    if (typeof param1 !== 'string') {
        throw new Error('param1 must be a string');
    }

    if (typeof param2 !== 'number') {
        throw new Error('param2 must be a number');
    }

    if (param2 < 0) {
        throw new Error('param2 must be non-negative');
    }

    // Implementation
    const result = {
        key1: `processed_${param1}`,
        key2: param2 * 2
    };

    return result;
}

module.exports = { functionName };

File: skills/10-stages-developing/templates/python/function-template.py
"""
Module for [module_name].

[Brief description of what this module does]
"""

from typing import List, Dict, Any, Optional


def function_name(param1: str, param2: int) -> Dict[str, Any]:
    """
    [Brief description of what the function does].

    Args:
        param1: [Description of param1]
        param2: [Description of param2]

    Returns:
        Dictionary containing:
        - key1: [Description of key1]
        - key2: [Description of key2]

    Raises:
        ValueError: If [condition for ValueError]
        TypeError: If [condition for TypeError]

    Examples:
        >>> function_name("example", 42)
        {'key1': 'value1', 'key2': 'value2'}
    """
    # Input validation
    if not param1:
        raise ValueError("param1 cannot be empty")

    if param2 < 0:
        raise ValueError("param2 must be non-negative")

    # Implementation
    result = {
        "key1": f"processed_{param1}",
        "key2": param2 * 2
    }

    return result
